### ISSUE TITLE
Feature request: Allow custom package manager instead of `npm install --save --save-dev -f --no-package-lock --prefix=`

### ISSUE THREAD

**Issue by shinebayar-g** [projen/projen#3675]
It seems `projen new --from` command [explicitly uses npm package manager](https://github.com/projen/projen/blob/5d763d73039d9e565173135df8062f47bb34dcfd/src/cli/util.ts#L59). 

https://github.com/projen/projen/blob/5d763d73039d9e565173135df8062f47bb34dcfd/src/cli/util.ts#L59

I guess we could technically use something like `npx projen new --from "git@github.com:my-org/my-repo.git"` to install from git. But we use yarn's [workspace support](https://yarnpkg.com/protocol/git#workspaces-support) to install specific package from the yarn workspace.
Command would look like this `npx projen new --from "git@github.com:my-org/my-repo.git#workspace=my-package"`

**Current behavior:**

```sh
Error: Command failed: npm install --save --save-dev -f --no-package-lock --prefix="/REDACTED" git@github.com:my-org/my-repo.git#workspace=my-package
npm warn using --force Recommended protections disabled.
npm error code 1
npm error The git reference could not be found
npm error command git --no-replace-objects checkout workspace=my-package
npm error error: pathspec 'workspace=my-package' did not match any file(s) known to git
```

**Comment by mrgrain** [comment]
This reads like `npm >= 7` supports workspaces as well.

<img width="643" alt="image" src="https://github.com/projen/projen/assets/379814/267ef9fd-7414-4b8e-8576-7c3afadfdbaf">

**Comment by mrgrain** [comment]
I agree this isn't ideal. On the other hand, this should be an implementation detail as `npm` supports plenty of ways to install packages (including from a tarball).

Whoever picks this up, please start with a design first. Supporting multiple package managers (and versions of them) is going to be maintenance nightmare. So I'd like to make sure the proposed design works.

**Comment by shinebayar-g** [comment]
> Your link implies that npm >= 7 supports workspaces as well.

Yeah. But while yarn is able to download specific workspace from npm/yarn monorepo, npm can't do the same.
